src/fielddef.ts

```diff
@@ -315,7 +316,7 @@ export function hasConditionalValueDef<F>(
 }
 
 export function isFieldDef<F>(
-  channelDef: ChannelDef<F>
+  channelDef: ChannelDef<F> | FieldDefBase<F>
```

Comment:
Why would FieldDefBase needs to be rechecked if it is a field def? This seems wrong.

Reply:
I added this so that I could use the `is*FieldDef` (e.g. `isTextFieldDef`, `isPositionFieldDef`) methods with a `FieldDefBase`. This is why I make the call to `isFieldDef` which you mention below.